pkg/processor/timeout/timeout.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2017 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package timeout
+
+import (
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/processor/trigger"
+
+	"github.com/nuclio/logger"
+)
+
+// Processor is minimal processor interface
+type Processor interface {
+	GetTriggers() []trigger.Trigger
+}
+
+// EventTimeoutWatcher checks for event timesout
+type EventTimeoutWatcher struct {
+	timeout   time.Duration
+	logger    logger.Logger
+	processor Processor
+}
+
+// NewEventTimeoutWatcher returns a new watcher
+func NewEventTimeoutWatcher(parentLogger logger.Logger, timeout time.Duration, processor Processor) *EventTimeoutWatcher {
+	watcher := &EventTimeoutWatcher{
+		logger:    parentLogger.GetChild("timeout"),
+		timeout:   timeout,
+		processor: processor,
+	}
+
+	go watcher.watch()
+
+	return watcher
+}
+
+func (w EventTimeoutWatcher) watch() {
+	for {
+		time.Sleep(w.timeout)
+		now := time.Now()
+		for triggerIndex, trigger := range w.processor.GetTriggers() {
+			for _, worker := range trigger.GetWorkers() {
```

Comment:
Let's parallelize this across workers so that a restart process takes less time. Can be done later, but we shouldn't forget to do this

Reply:
Added TODO in the code